### PR TITLE
Limit overhead of capturing stacktrace in REST Client

### LINKED ISSUE #42508: Explore the use of StackWalker in REST Client
### Description

Currently in the REST Client when `quarkus.rest-client.capture-stacktrace` is set to true  (which is the default), we capture the entire stacktrace of the Thread in order to enhance the debugging experience when something goes wrong.

@johnaohara has found however that in some cases up to 20% of the users application code callstack in capturing a stack trace, which is a waste of cpu cycles, but also forces the JVM to safepoint the mutating threads during every call to the api

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by geoand**
This is done by utilizing the StackWalker API
and limiting the number of frames captured

- Closes: #42508

### CODE REVIEW

**Comment by geoand** [comment]
That's the only way to produce a meaningful stacktrace when an error occurs

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientCaptureCurrentContextRestHandler.java [hunk 1]
```diff
@@ -35,21 +37,18 @@ public void handle(RestClientRequestContext requestContext) throws Exception {
...
+        List<StackTraceElement> effectiveStackTrace = StackWalker.getInstance().walk(new Function<>() {
+            @Override
+            public List<StackTraceElement> apply(Stream<StackWalker.StackFrame> stream) {
+                return stream.dropWhile(new Predicate<>() {
+                    @Override
+                    public boolean test(StackWalker.StackFrame stackFrame) {
+                        String className = stackFrame.getClassName();
+                        return className.startsWith(RESTEASY_REACTIVE_PACKAGE) || className.contains(AUTOGENERATED_TAG);
+                    }
+                }).limit(5).map(StackWalker.StackFrame::toStackTraceElement).toList();
```
Comment (brunobat) [inline independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientCaptureCurrentContextRestHandler.java:49]: Is 5 lines enough?
this probably needs to be documented or add a config to keep the switch to the old behaviour.

Comment (geoand) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientCaptureCurrentContextRestHandler.java:49]: It's fine for identifying which REST Client is causing the problem and where it was called from.

It's not a complete stacktrace, but the complete one is mostly useless except for the top frames

**Comment by geoand** [comment]
Let's close this

**Pull request closed without merging**